ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -1093,21 +1093,23 @@ private void checkDataIntegrity() throws StoreException {
   }
 
   /**
-   * Gets all the entries upto maxEntries from the start of a given key (exclusive) or all entries if key is null,
-   * till maxTotalSizeOfEntriesInBytes
+   * Gets all the entries upto maxEntries from the start of a given key (whether to include this key depends on variable
+   * inclusive) or all entries if key is null, till maxTotalSizeOfEntriesInBytes
    * @param key The key from where to start retrieving entries.
    *            If the key is null, all entries are retrieved upto max entries
    * @param findEntriesCondition The condition that determines when to stop fetching entries.
    * @param entries The input entries list that needs to be filled. The entries list can have existing entries
    * @param currentTotalSizeOfEntriesInBytes The current total size in bytes of the entries
+   * @param inclusive whether to include the entry associated with given key in returned result.
    * @return true if any entries were added.
    * @throws StoreException
    */
   boolean getEntriesSince(StoreKey key, FindEntriesCondition findEntriesCondition, List<MessageInfo> entries,
```

Comment:
Can we hav a test for `inclusive` true case?

Reply:
Thanks for bringing this up, we actually covered both true and false in `rebuildTokenBasedOnResetKeyTest` in the `IndexTest.java`. See below log (I printed out `inclusive` value).
```
first key = 2CC512DBRB
random key = SYNJSPHQ3I
[2021-03-18 15:32:50,630] INFO Revalidated token version: 2 type: IndexBased incarnationId 66309b30-e405-4fdf-a956-532e9d40c1b0 inclusiveness false sessionId ee0f30ef-6832-48dc-bfd2-d8fd7c7f1184 storeKey SYNJSPHQ3I resetKey SYNJSPHQ3I resetKeyType PUT resetKeyVersion 0 offset Name = [0_0] Offset = [17] bytesRead -1 because it is invalid for the index segment map (com.github.ambry.store.PersistentIndex)
Inclusive = true
[2021-03-18 15:32:50,632] INFO Revalidated token version: 2 type: JournalBased incarnationId 66309b30-e405-4fdf-a956-532e9d40c1b0 inclusiveness false sessionId ee0f30ef-6832-48dc-bfd2-d8fd7c7f1184 resetKey N2YCGV1UB7 resetKeyType PUT resetKeyVersion 0 offset Name = [2_1] Offset = [365] bytesRead -1 because it is invalid for the index segment map (com.github.ambry.store.PersistentIndex)
Inclusive = true
[2021-03-18 15:32:50,633] INFO Reset key NRP94DG8E1 is not found in current index, token is reset to the beginning of whole log (com.github.ambry.store.PersistentIndex)
[2021-03-18 15:32:50,633] INFO Revalidated token version: 2 type: JournalBased incarnationId 66309b30-e405-4fdf-a956-532e9d40c1b0 inclusiveness false sessionId ee0f30ef-6832-48dc-bfd2-d8fd7c7f1184 resetKey NRP94DG8E1 resetKeyType PUT resetKeyVersion 0 offset Name = [2_1] Offset = [365] bytesRead -1 because it is invalid for the index segment map (com.github.ambry.store.PersistentIndex)
Inclusive = false
```